ResearchKit/Common/ORKPieChartView.m

```diff
@@ -0,0 +1,543 @@
+    _shouldAnimateLegend = YES;
+    _animationDuration = 0.35f;
+    _shouldDrawClockwise = YES;
+    
+    _actualValues = [NSMutableArray new];
+    _normalizedValues = [NSMutableArray new];
+    _sumOfValues = 0;
+
+    _titleLabel = [UILabel new];
+    _textLabel = [UILabel new];
+    [_titleLabel setTextAlignment:NSTextAlignmentCenter];
+    [_textLabel setTextAlignment:NSTextAlignmentCenter];
+    
+    _emptyText = NSLocalizedString(@"No Data", @"No Data");
+    
+    _emptyLabel = [UILabel new];
+    _emptyLabel.text = _emptyText;
+    _emptyLabel.textAlignment = NSTextAlignmentCenter;
+    
+    _legendLayout = [[ORKCentredCollectionViewLayout alloc] init];
+    _legendView = [[UICollectionView alloc]initWithFrame:CGRectZero collectionViewLayout: _legendLayout];
+    [_legendView registerClass: NSClassFromString(@"ORKLegendCollectionViewCell") forCellWithReuseIdentifier:@"cell"];
+    _legendView.backgroundColor = [UIColor clearColor];
+    _legendView.translatesAutoresizingMaskIntoConstraints = NO;
+    _legendView.dataSource = self;
+    _legendView.delegate = self;
+    
+    _plotView = [UIView new];
+    _plotView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    _contentView = [UIView new];
+    _contentView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    [self addSubview: _contentView];
+    [_contentView addSubview:_plotView];
+    [_contentView addSubview:_legendView];
+    
+    [self updateAppearance];
+}
+
+#pragma mark - Layout
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    
+    // legendView height
+    [_contentView layoutIfNeeded];
+    CGFloat legendHeight = [_legendView.collectionViewLayout collectionViewContentSize].height;
+    _legendView.frame = CGRectMake(0, 0, CGRectGetWidth(self.bounds), legendHeight);
+    
+    // plotView height
+    CGFloat remainingHeight = CGRectGetHeight(_contentView.frame) - legendHeight;
+    CGFloat plotHeight = MIN(remainingHeight, _contentView.frame.size.width); // The largest diameter possible within the bounds.
+    
+    // padding height
+    CGFloat contentHeight = legendHeight + plotHeight;
+    CGFloat verticalPadding = (CGRectGetHeight(_contentView.frame) - contentHeight) / 2;
+    CGFloat plotLegendPadding = 8;
+    
+    // vertically centered frames
+    _plotView.frame = CGRectMake(0, verticalPadding - (plotLegendPadding * 0.5), CGRectGetWidth(_contentView.frame), plotHeight);
+    _legendView.frame = CGRectMake(0, verticalPadding + plotHeight + (plotLegendPadding * 0.5), CGRectGetWidth(_contentView.frame), legendHeight);
+    
+    // circle path
+    CGFloat startAngle = _originAngle;
+    CGFloat endAngle = startAngle + (2 * M_PI);
+    CGFloat unlabelledRadius = plotHeight * 0.5;
+    CGFloat labelHeight = [@"100%" boundingRectWithSize: CGRectInfinite.size options:0 attributes:@{NSFontAttributeName : _percentageFont} context:nil].size.height;
+    CGFloat labelledRadius = unlabelledRadius - (labelHeight + _percentageLabelOffset);
```

Comment:
Let's use `labeled...`/`unlabeled...`, which is a more standard American English spelling.